app/services/validators/session-validator.js

```diff
@@ -0,0 +1,215 @@
+const validator = require('./common-validator')
+const VALIDATION_ERROR = new Error('An error has occured')
+const decrypt = require('../../services/helpers/decrypt')
+const dateFormatter = require('../../services/date-formatter')
+const claimTypeEnum = require('../../constants/claim-type-enum')
+
+class sessionValidator {
+  static validateData (session) {
+    if (session['dobEncoded'] && session['dobEncoded'] != null) {
+      var dob = dateFormatter.decodeDate(session['dobEncoded'])
+
+      this.validateParam(dob, 'isValidDateOfBirth')
+    }
+
+    if (session['relationship'] && session['relationship'] != null) {
+      var relationship = session['relationship']
+
+      this.validateParam(relationship, 'isValidPrisonerRelationship')
+    }
+
+    if (session['benefit'] && session['benefit'] != null) {
+      var benefit = session['benefit']
+
+      this.validateParam(benefit, 'isValidBenefit')
+    }
+
+    if (session['referenceId'] && session['referenceId'] != null) {
+      var referenceId = decrypt(session['referenceId'])
+
+      this.validateParam(referenceId.reference, 'isValidReference')
+      this.validateParam(referenceId.id, 'isValidReferenceId')
+    }
+
+    if (session['decryptedRef'] && session['decryptedRef'] != null) {
+      var reference = session['decryptedRef']
+
+      this.validateParam(reference, 'isValidReference')
+    }
+
+    if (session['claimType'] && session['claimType'] != null) {
+      var claimType = session['claimType']
+
+      this.validateParam(claimType, 'isValidClaimType')
+    }
+
+    if (session['advanceOrPast'] && session['advanceOrPast'] != null) {
+      var advanceOrPast = session['advanceOrPast']
+
+      this.validateParam(advanceOrPast, 'isValidAdvanceOrPast')
+    }
+
+    if (session['claimId'] && session['claimId'] != null) {
+      this.validateParam(session['claimId'].toString(), 'isNumeric')
+    }
+
+    if (session['claimDocumentId'] && session['claimDocumentId'] != null) {
+      this.validateParam(session['claimDocumentId'].toString(), 'isNumeric')
+    }
+  }
+
+  static validateParam (param, validateFunction) {
+    if (param) {
+      if (!validator[validateFunction](param)) {
+        throw VALIDATION_ERROR
+      }
+    }
+  }
+
+  static getErrorPath (session, url) {
```

Comment:
expose this function separately so routes can call to get this redirect path instead of needing to return an object with unclear elements:
e.g.  
```
if (!sessionHandler.validateSessionForPath(req.session, req.url)) {
  return res.redirect(sessionHandler.getErrorPath(req.session, req.url))
}
```